### PR TITLE
Refactor: move some files to @grafana/data

### PR DESCRIPTION
**Pull request by ryantxu**
I have been looking at lots of plugins recently.  It would be nice to start migrating some of the existing plugins to use `@grafana/` packages.

The most widely used functions (that require hoops for testing!) are `datemath` and `kbn`

This PR aims to move some obvious stuff to `@grafana/data` *hopefully* before 6.3 is released so that package names can be consistent

### CODE REVIEW

File: packages/grafana-data/package.json [hunk 1]
```diff
@@ -33,6 +34,7 @@
     "rollup-plugin-terser": "4.0.4",
     "rollup-plugin-typescript2": "0.19.3",
     "rollup-plugin-visualizer": "0.9.2",
+    "sinon": "1.17.6",
```
Comment (torkelo) [inline packages/grafana-data/package.json]: do not think we need this, is only one test that use it I think, should use jest.fn() instead

**Comment by ryantxu** [comment]
@dprokop -- I got the errors down to one... and my brain must be fried, if anythign is obvious to you, let me know!
```
>> FAIL public/app/features/explore/utils/ResultProcessor.test.ts
>> ● Test suite failed to run
>> 
>>     TypeError: Cannot read property 'none' of undefined
>> 
>>       50 |   showingGraph: true,
>>       51 |   showingLogs: true,
>>     > 52 |   dedupStrategy: LogsDedupStrategy.none,
>>          |                                    ^
>>       53 | };
>>       54 | 
>>       55 | const MAX_HISTORY_ITEMS = 100;
>> 
>>       at Object.<anonymous> (public/app/core/utils/explore.ts:52:36)
>>       at Object.<anonymous> (public/app/features/explore/utils/ResultProcessor.ts:8:1)
>>       at Object.<anonymous> (public/app/features/explore/utils/ResultProcessor.test.ts:16:1)
```

**Comment by ryantxu** [comment]
My issue seems related to how `dateTime` gets mocked in that test.  Previously:
```
jest.mock('@grafana/ui/src/utils/moment_wrapper', () => ({
  dateTime: (ts: any) => {
    return {
      valueOf: () => ts,
      fromNow: () => 'fromNow() jest mocked',
      format: (fmt: string) => 'format() jest mocked',
    };
  },
  toUtc: (ts: any) => {
    return {
      format: (fmt: string) => 'format() jest mocked',
    };
  },
}));
```

File: packages/grafana-data/package.json (hunk 1 above)
Comment (ryantxu) [reply packages/grafana-data/package.json]: I agree... but it is not a trivial change.  The test is using sinon for: `SinonFakeTimers`,  `clock = sinon.useFakeTimers(unix);`

Jest has `jest.useFakeTimers();` this does all the same things for interval, but does not let you reset the global clock.  Reading the jest docs on it I found https://github.com/taitulism/jest-fake-timers-issue-demo.

This test could be refactored to only use jest, but will require understanding the test a bit more and should not hold up this PR.

**Comment by ryantxu** [comment]
@torkelo -- sorry for the 250 file PR!  

There are no logic changes, only moving files to `@grafana/data` and updating import paths.

The one thing that had me stumped was the [import for this test.](https://github.com/grafana/grafana/pull/17952/files#diff-0315afdbadca215c057055152b86469aR1)

**Comment by torkelo** [comment]
Looks good! ready for merge?

**Review by torkelo — APPROVED** [review]

**Pull request merged**